pkg/config/agent.go

```diff
@@ -15,11 +15,16 @@ func ValidateAgentImage() error {
 }
 
 func GetParsedAgentImage() (string, string, string) {
-	parts := strings.Split(AgentImageName, ":")
+	imgParts := strings.Split(AgentImageName, "/")
+	registry := strings.Join(imgParts[0:len(imgParts)-1], "/")
```

Comment:
defense nil panic and add test cases for the empty name

Reply:
Is it fine to change the function to `GetParsedAgentImage() (string, string, string, error)` without panic?

Reply:
Fixed.